File: Java/src/test/java/StringCalculatorShould.java
import com.codurance.stringcalculator.Calculator;
import org.junit.Test;

import static org.hamcrest.CoreMatchers.is;
import static org.junit.Assert.assertThat;

public class StringCalculatorShould {

    @Test
    public void add_TakesEmptyString_ReturnsZero() {
        Calculator calculator = new Calculator();

        int result = calculator.add("");

        assertThat(result, is(0));
    }

    @Test
    public void add_TakesSingleDigit_ReturnsSingleDigit() {
        Calculator calculator = new Calculator();

        int result = calculator.add("4");

        assertThat(result, is(4));
    }

    @Test
    public void add_TakesTwoDigits_AddsAndReturnsSingleDigit() {
        Calculator calculator = new Calculator();

        int result = calculator.add("1,2");

        assertThat(result, is(3));
    }

    @Test
    public void add_TakesArbitraryNumberOfDigits_AddsAndReturnsSingleDigit() {
        Calculator calculator = new Calculator();

        int result = calculator.add("1,2,3");

        assertThat(result, is(6));
    }
}

File: C#/StringCalculator/StringCalculator.UnitTests/StringCalculatorShould.cs
using Xunit;

namespace StringCalculator.UnitTests
{
	public class StringCalculatorShould
	{
		[Fact]
		public void Add_TakesEmptyString_ReturnsZero()
		{
			var calculator = new Calculator();

			var result = calculator.Add(string.Empty);

			Assert.Equal(0, result);
		}

		[Fact]
		public void Add_TakesSingleDigit_ReturnsSingleDigit()
		{
			var calculator = new Calculator();

			var result = calculator.Add("4");

			Assert.Equal(4, result);
		}

		[Fact]
		public void Add_TakesTwoDigits_AddsAndReturnsSingleDigit()
		{
			var calculator = new Calculator();

			var result = calculator.Add("1,2");

			Assert.Equal(3, result);
		}

		[Fact]
		public void Add_TakesArbitraryNumberOfDigits_AddsAndReturnsSingleDigit()
		{
			var calculator = new Calculator();

			var result = calculator.Add("1,2,3");

			Assert.Equal(6, result);
		}
	}
}